.github/workflows/docker.yml

```diff
@@ -31,16 +31,16 @@ jobs:
         id: gitversion
         uses: gittools/actions/gitversion/execute@v0.9.7
       -
-        name: Publish release on GitHub
-        uses: softprops/action-gh-release@v1
+        name: Create tag for GitHub
```

Comment:
@merry-degaga my thought is that if we do a patch then we should only create tag and update the unreleased tag of docker image but if we do minor or major release then we should create a release, update the docker image tag of latest as well as unreleased.


example: say in one scenario: we want to do a small change to our project and that small change should only create a newer github tag, same tag for docker image and update the unreleased tag.

in another scenario, we want to do a minor or major change or release all the earlier patch changes so now we want a newer github tag, a newer github release, update the latest docker image tag as well as unreleased docker image tag.

is that possible? we are good even if we have another workflow that gets triggered for 2nd scenario.

Reply:
Okay I did find two ways of doing this, which would be most preferred? The first method will  automate the docker build/deploy for a manually published release. The second way will automate release creation and docker build/deploy but needs to be manually triggered. In either case, a new workflow yml file will be created.

Workflow 1
1. Auto triggers when a release is manually published 
2. Using github context, it will get the tag attached to the published release
3. This tag and the 'latest' tag attached to a built and deployed docker image

Workflow 2:
1. Manual trigger workflow using workflow_dispatch
2. Uses GitHub action https://github.com/softprops/action-gh-release to automate publish of release
3. This will automate the name creation and setting the tag_name for the release and publish the release
4. Docker image will be built and deployed with 'latest' tag and tag_name
5. Description for the release can be manually edited later

Reply:
I think workflow 2 should work for us.